core/watcher/eventqueue/uuid.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package eventqueue
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/juju/collections/transform"
+	"github.com/juju/errors"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/core/changestream"
+)
+
+// UUIDWatcher watches for changes to a database table.
+// Any time rows change in the watched table, the changed UUIDs are emitted.
+type UUIDWatcher struct {
+	*BaseWatcher
+
+	out       chan []string
+	tableName string
+	selectAll string
+}
+
+// NewUUIDWatcher returns a new watcher that receives changes from the
+// input base watcher's db/queue when rows in the input table change.
+func NewUUIDWatcher(base *BaseWatcher, tableName string) *UUIDWatcher {
+	w := &UUIDWatcher{
+		BaseWatcher: base,
+		out:         make(chan []string),
+		tableName:   tableName,
+		selectAll:   fmt.Sprintf("SELECT uuid FROM %s", tableName),
```

Comment:
I wonder at the very least this should be `%q` as this might contain spaces or hyphens.

Reply:
Tables and views in the Juju database will not contain spaces or hyphens. No debate will occur on this matter.

Reply:
Currently, we're using singular, lowercase with underscore table names https://github.com/juju/juju/blob/main/domain/schema/controller_test.go#L56-L79